packages/llamaindex/src/tools/AzureDynamicSessionTool.ts

```diff
@@ -0,0 +1,355 @@
+import {
+  DefaultAzureCredential,
+  getBearerTokenProvider,
+} from "@azure/identity";
+import { getEnv } from "@llamaindex/env";
+import crypto from "node:crypto";
+import { createWriteStream } from "node:fs";
+import { readFile } from "node:fs/promises";
+import { dirname, join, resolve } from "node:path";
+import { Readable } from "node:stream";
+import { fileURLToPath } from "node:url";
+import { FormData, fetch } from "undici";
```

Comment:
isn't FormData part of node?

Reply:
Yes, native FormData is supported since node 16. This `FormData` patch from `undici` adds compatibility with `undici.request`([see PR](https://github.com/nodejs/undici/pull/1332)).

Reply:
i see, and we can't use normal fetch without undici?

Reply:
I wasn't sure about `fetch` support in other runtimes. Node.js introduces `fetch` API in v18 and it's backed by `undici`. Happy to move to global `fetch` if this is compatible with other runtimes.

Reply:
use global fetch, we support node >18, not 16

Reply:
done.